src/main/java/org/webshar/hrms/repository/LeaveAllocationRepository.java

```diff
@@ -19,36 +19,20 @@
       @Param("end_date") final Date endDate
   );
 
-  public List<LeaveAllocation> findAllByEmployeeIdAndLeaveTypeIdAndStartDateGreaterThanEqualAndEndDateLessThanEqual(
-      @Param("employee_id") final Long employeeId,
-      @Param("leave_type_id") final Long leaveTypeId,
+  public List<LeaveAllocation> findAllByStartDateGreaterThanEqualAndStartDateLessThanEqual(
       @Param("start_date") final Date startDate,
       @Param("end_date") final Date endDate
```

Comment:
see if you can use Java 8 dates APIs e.g. `LocalDate` for `startDate` and `endDate` modeling which can help in implicitly avoiding some regression issues related to `Date` objects.